documentation/site/pages/developers/tutorials/request.mdx

```diff
@@ -101,45 +248,86 @@ let input_url = client.upload_input(&input_bytes).await?;
+### Onchain vs Offchain
+
+The Boundless protocol allows you to submit requests both onchain and offchain.
 
-### 4. Execute the Program to Estimate Cycles and to Get the Journal
+#### Onchain
+
+To submit a request onchain, we use:
+
+<StripRustCodeComments>
+```rust
+# use anyhow::Result;
+# use boundless_market::{Client};
+# async fn create_proof_request(
+# client: Client,
+# program: &'static [u8],
+# input: &[u8]) -> Result<()> {
+// Create a request using new_request
+let request = client.new_request().with_program(ECHO_ELF).with_stdin(echo_message.as_bytes());
 
-Before sending a proof request, developers should execute the program. This achieves a few separate goals:
+// Submit the request directly
+let (request_id, expires_at) = client.submit_onchain(request).await?; // [!code hl] // [!code focus]
+# Ok(())
+# }
+```
+</StripRustCodeComments>
 
-- It estimates the number of cycles in the program, which can be used to estimate the cost of the proof request offer.
-- It makes sure that the program executes without errors, saving the developer money by not sending unprovable proof requests to the market.
-- It retrieves the journal, the public outputs of the program, which are used to check proof validity along with the imageID.
+#### Offchain
 
-To execute the program and retrieve the cycle count (in MCycles), and journal:
+:::note[Sumbitting offchain?]
+When using offchain requests, you are required to deposit funds into the Boundless market contract before you can make any proof requests. This can be done with the [Boundless CLI deposit subcommand](/developers/tooling/cli#1-deposit).
+:::
+
+To submit a request offchain, we use:
 
 <StripRustCodeComments>
 ```rust
-# use risc0_zkvm::{default_executor, ExecutorEnv};
-# fn execute_program(input_bytes: Vec<u8>, program: &[u8]) -> Result<(), Box<dyn std::error::Error>> {
-let env = ExecutorEnv::builder().write_slice(&input_bytes).build()?;
-let session_info = default_executor().execute(env, program)?;
-let mcycles_count = session_info
-  .segments
-  .iter()
-  .map(|segment| 1 << segment.po2)
-  .sum::<u64>()
-  .div_ceil(1_000_000);
-let journal = session_info.journal;
+# use anyhow::Result;
+# use boundless_market::{Client};
+# async fn create_proof_request(
+# client: Client,
+# program: &'static [u8],
+# input: &[u8]) -> Result<()> {
+// Create a request using new_request
+let request = client.new_request().with_program(ECHO_ELF).with_stdin(echo_message.as_bytes());
+
+// Submit the request directly
+let (request_id, expires_at) = client.submit_offchain(request).await?; // [!code hl] // [!code focus]
 # Ok(())
 # }
 ```
 </StripRustCodeComments>
 
-> Relevant links: [ExecutorEnvBuilder](https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.ExecutorEnvBuilder.html)
+### Offer Parameters
```

Comment:
So, in addition to `OfferParams` that can be set when building the client, there are also config options when building the `Client`. In particular, if someone wants to use the cycle count to decide the price, then they can do so by calling `Client::builder().config_offer_layer(|config| /* set options here */)`. Could you also show this? Sorry that it's a bit confusing that there are two ways to do it. The main difference is that you use the methods on the client builder when you want to adjust the logic the builder uses, you use `OfferParams` when you want to override the builder.

https://github.com/boundless-xyz/boundless/blob/1f2019496da6745363dee9f59d178a8dc151eede/crates/boundless-market/src/client.rs#L286-L305

Reply:
@nategraf could you please take a look here and let me know if this resolves your comment here:

https://boundless-documentation-g273n87e4-risczero.vercel.app/developers/tutorials/request#offer